src/main/webapp/forms/placeGuideForm.js

```diff
@@ -137,3 +138,36 @@ function fillFormWithPlaceGuideData(placeGuide) {
       DUMMY_DATA_FOR_PLACE_NAME);
 }
 
+/**
+ * Will fill the form if query string exists. Query string exists only when user wants
+ * to edit a place guide.
+ */
+function fillFormWithPlaceGuideToEdit() {
+  if (window.location.search != '') {
+    const GET = {};
+    const queryString = decodeURI(window.location.search.replace(/^\?/, ''));
+    queryString.split(/\&/).forEach(function(keyValuePair) {
+      const paramName = keyValuePair.replace(/=.*$/, '');
+      const paramValue = keyValuePair.replace(/^[^=]*\=/, '');
+      GET[paramName] = paramValue;
+    });
+    document.getElementById('id').value = GET['placeGuideId'];
+    document.getElementById('placeId').value = GET['placeId'];
+    document.getElementById('name').value = GET['name'];
+    document.getElementById('audioPlayer').src = getBlobSrc(GET['audioKey']);
+    if (GET['imageKey'] != 'undefined') {
+      document.getElementById('imagePreview').src = getBlobSrc(GET['imageKey']);
+    }
+    document.getElementById('imagePreview').style.display = 'block';
+    if (GET['description'] != 'undefined') {
+      document.getElementById('description').value = GET['description'];
+    }
+    document.getElementById('length').value = GET['length'];
+    document.getElementById('isPublic').value = GET['isPublic'];
+    if (GET['placeName'] != 'null') {
+      document.getElementById('placeName').value = GET['placeName'];
+    }
+    document.getElementById('latitude').value = GET['latitude'];
+    document.getElementById('longitude').value = GET['longitude'];
+  }
+}
```

Comment:
Don't you need to use the MDC objects(as initialised above) for setting the values? Are the text input labels, textarea labels, etc. working correctly if you modify directly the value of the inputs instead of setting the value of the MDC objects?

Reply:
Used the MDC objects for setting the values.